File: src/publish_img.py
#!/usr/bin/env python
import rospy
import roslib 
from sensor_msgs.msg import Image
from std_srvs.srv import SetBool
from geometry_msgs.msg import PoseStamped, TransformStamped
from gazebo_msgs.srv import *
import math

# Because of transformations
import tf_conversions
import tf2_ros

class ImgPublisher:
    def __init__(self):
        self.counter = 0
        self.image = None
        self.sub = rospy.Subscriber('/camera/rgb/image_rect_color',Image, self.call_receive_rgb_img)
        self.pub = rospy.Publisher('/custom_img', Image, queue_size=1)
        self.reset_service = rospy.Service('/reset_counter', SetBool, self.call_reset_counter)

    def call_receive_rgb_img(self, msg):
        self.counter = self.counter + 1
        print(self.counter)
        # self.image = msg.data
        self.pub.publish(msg)
    
    def call_reset_counter(self, req):
        if req.data:
            self.counter = 0
            return True, 'Counter reset'
        return False, 'Counter not reset'

if __name__ == "__main__":
    rospy.init_node('img_publisher', anonymous=True)
    ImgPublisher()
    rospy.spin()

File: src/publish_img.cpp
#include <ros/ros.h>
#include <sensor_msgs/Image.h>
#include <std_srvs/SetBool.h>


class ImgPublisher {

    private:
    int counter;
    ros::Publisher pub;
    ros::Subscriber sub;
    ros::ServiceServer reset_service;

    public:
    ImgPublisher(ros::NodeHandle *nh){
        counter = 0;
        // image = 0;
        sub = nh-> subscribe("/camera/rgb/image_rect_color",1, 
            &ImgPublisher::call_receive_rgb_img, this);

        pub = nh-> advertise<sensor_msgs::Image>("/custom_img",1);

        reset_service = nh->advertiseService("/reset_counter",
            &ImgPublisher::call_reset_counter, this);

    }

    void call_receive_rgb_img(const sensor_msgs::Image img){
        counter += 1;
        // cout << counter;
        ROS_INFO("Message received:");
        pub.publish(img);
    }

    bool call_reset_counter(std_srvs::SetBool::Request &req,
                            std_srvs::SetBool::Response &res)
    {
        if(req.data){
            counter = 0;
            res.success = true;
            res.message = "Counter reset";
        }else{
            res.success = false;
            res.message = "Counter NOT reset";
        }
        return true;
    }

};

int main (int argc, char **argv){
    ros::init(argc, argv, "publish_img");
    // ros::NodeHandle nh;
    ROS_INFO(" test:");
    // ImgPublisher ip = ImgPublisher(&nh);
    // ros::spin();
}